src/platform/plugins/shared/controls/public/controls/esql_control/get_esql_control_factory.tsx

```diff
@@ -54,11 +55,18 @@ export const getESQLControlFactory = (): ControlFactory<ESQLControlState, ESQLCo
       };
 
       function serializeState() {
+        const latestState = {
+          ...defaultControlManager.getLatestState(),
+          ...selections.getLatestState(),
+        };
+
+        const rawState =
```

Comment:
Can this logic be moved into `selections.getLatestState`?

Reply:
You're right, it actually does still work if I do this at the `getLatestState` level. Doesn't require any refactoring. Pushing a fix